### ISSUE TITLE
Alert manager sending 2 messages sent for every alert "alerting"

### ISSUE THREAD

**Issue by tfiduccia** [rancher/rancher#19058]
Version - 2.2

**Steps to reproduce (least amount of steps as possible):**
1.  Create cluster
2.  Go to cluster and set up slack notifier
3.  Go to Alerts and create a Alert Group pointed to the slack notifier.  Make the critical Node rule where the CPU reservation is over 5%.  (Alert should fire)

**Result:**  Two critical alerts get sent from the alert manager.  It doesn't seem to matter what type of Alert rule you set or where you send it to, it always sends 2.

![image](https://user-images.githubusercontent.com/11514927/54724869-21d7a100-4b2a-11e9-8b1b-8aec704ab6ce.png)

**Comment by aiwantaozi** [comment]
Hi, I can't reproduce in the master, how many nodes for this cluster, any extra steps for the node?

**Comment by aiwantaozi** [comment]
@tfiduccia Could you help check again?

**Comment by davidnuzik** [comment]
Version: Master (4/1/19) ImageID: e59026e98f6e
I am not able to reproduce as well in the latest master. I tried a couple of different configurations: 1 node scale up to 2 then 3 with ALL ROLES, 1 master+CP node and 3 worker nodes. 

In either case I was not able to reproduce on a fresh install.
Alert rule:
![image](https://user-images.githubusercontent.com/45179589/55350693-828fa380-5471-11e9-8717-c31a1c977859.png)

**Comment by tfiduccia** [comment]
@aiwantaozi - I wasn't able to reproduce with node alert, but I'm still getting two for pod alerts.  This for a DO cluster with 3 machines.  I set alert at cluster level.  It's a simple pod alert.
![image](https://user-images.githubusercontent.com/11514927/55352164-1616a380-5475-11e9-974d-79458d2e5063.png)

**Comment by aiwantaozi** [comment]
Hi @tfiduccia Event is very special, you could check ``show more``` whether the message is different, if the message is different will send two alerts.

**Comment by tfiduccia** [comment]
@aiwantaozi - Both alerts get sent at the same time and the repeat interval is set for 60 mins.  In the show more:
The first alert has the two event messages (assigned & pull image)
The second alert has the first two event (assigned & pull image), then also has 3 more (create, start and successfully pulled)

It's not just for pod create either.  I get the same type of thing for when I delete the workld.  Two Alerts get sent.

Single rancher server - setup on GCE machine.  DO or ec2 RKE cluster.

**Comment by aiwantaozi** [comment]
@tfiduccia Linked pr have updated the event alert to only send current event. Will update after merged.

**Comment by tfiduccia** [comment]
@aiwantaozi - I have no idea what is going on with this issue.  It has stayed in ready-for-review, but not sure who is supposed to review it and how it's supposed to get into the master build.  Can you please follow up with @loganhz or @westlywright

**Comment by aiwantaozi** [comment]
Sure, I will check with Logan.

**Comment by davidnuzik** [comment]
@aiwantaozi I am not able to get alerts to work for me on v2.2.3-rc5 or latest master.
It may or may not be related to this issue though -- but I notice the alert manager seems to fail to deploy? May I ask if you happen to run into this issue as well?

Here is a snippet of my logs grep'ed for "error"
```
2019/05/03 18:14:34 [WARNING] error updating ns p-f2lgt status: Operation cannot be fulfilled on namespaces "p-f2lgt": the object has been modified; please apply your changes to the latest version and try again                                                           
2019/05/03 18:14:34 [ERROR] ClusterAlertRuleController c-665qj/car-6ff8s [cluster-alert-rule-deployer] failed with : failed to get Alertmanager Deployment information, statefulsets.apps "alertmanager-cluster-alerting" not found, [cluster-alert-rule-controller] failed with : Failed to get service for alertmanager, service "cattle-prometheus/alertmanager-operated" not found
2019/05/03 18:14:39 [ERROR] ProjectAlertGroupController p-f2lgt/projectalert-workload-alert [project-alert-group-deployer] failed with : failed to get Alertmanager Deployment information, statefulsets.apps "alertmanager-cluster-alerting" not found, [project-alert-group-controller] failed with : Failed to get service for alertmanager, service "cattle-prometheus/alertmanager-operated" not found
2019/05/03 18:14:39 [ERROR] ProjectAlertRuleController p-f2lgt/memory-close-to-resource-limited [project-alert-rule-deployer] failed with : failed to get Alertmanager Deployment information, statefulsets.apps "alertmanager-cluster-alerting" not found, [project-alert-rule-controller] failed with : Failed to get service for alertmanager, service "cattle-prometheus/alertmanager-operated" not found
2019/05/03 18:14:39 [ERROR] ProjectAlertRuleController p-f2lgt/less-than-half-workload-available [project-alert-rule-deployer] failed with : failed to get Alertmanager Deployment information, statefulsets.apps "alertmanager-cluster-alerting" not found, [project-alert-rule-controller] failed with : Failed to get service for alertmanager, service "cattle-prometheus/alertmanager-operated" not found
2019/05/03 18:14:44 [ERROR] ProjectAlertGroupController p-f2lgt/projectalert-workload-alert [project-alert-group-deployer] failed with : failed to get Alertmanager Deployment information, statefulsets.apps "alertmanager-cluster-alerting" not found, [project-alert-group-controller] failed with : Failed to get service for alertmanager, service "cattle-prometheus/alertmanager-operated" not found
2019/05/03 18:14:44 [ERROR] ProjectAlertRuleController p-f2lgt/less-than-half-workload-available [project-alert-rule-deployer] failed with : failed to get Alertmanager Deployment information, statefulsets.apps "alertmanager-cluster-alerting" not found, [project-alert-rule-controller] failed with : Failed to get service for alertmanager, service "cattle-prometheus/alertmanager-operated" not found
2019/05/03 18:14:44 [ERROR] ProjectAlertRuleController p-f2lgt/memory-close-to-resource-limited [project-alert-rule-deployer] failed with : failed to get Alertmanager Deployment information, statefulsets.apps "alertmanager-cluster-alerting" not found, [project-alert-rule-controller] failed with : Failed to get service for alertmanager, service "cattle-prometheus/alertmanager-operated" not found
2019/05/03 18:14:49 [ERROR] ProjectAlertGroupController p-f2lgt/projectalert-workload-alert [project-alert-group-deployer] failed with : failed to get Alertmanager Deployment information, statefulsets.apps "alertmanager-cluster-alerting" not found, [project-alert-group-controller] failed with : Failed to get service for alertmanager, service "cattle-prometheus/alertmanager-operated" not found
2019/05/03 18:14:49 [ERROR] ProjectAlertRuleController p-f2lgt/less-than-half-workload-available [project-alert-rule-deployer] failed with : failed to get Alertmanager Deployment information, statefulsets.apps "alertmanager-cluster-alerting" not found, [project-alert-rule-controller] failed with : Failed to get service for alertmanager, service "cattle-prometheus/alertmanager-operated" not found
2019/05/03 18:14:49 [ERROR] ProjectAlertRuleController p-f2lgt/memory-close-to-resource-limited [project-alert-rule-deployer] failed with : failed to get Alertmanager Deployment information, statefulsets.apps "alertmanager-cluster-alerting" not found, [project-alert-rule-controller] failed with : Failed to get service for alertmanager, service "cattle-prometheus/alertmanager-operated" not found
2019/05/03 18:14:50 [ERROR] Failed to send alert: Failed to get service for alertmanager, service "cattle-prometheus/alertmanager-operated" not found                                                                                                                        
2019/05/03 18:14:50 [ERROR] Failed to send alert: Failed to get service for alertmanager, service "cattle-prometheus/alertmanager-operated" not found                                                                                                                        
2019/05/03 18:14:50 [ERROR] Failed to send alert: Failed to get service for alertmanager, service "cattle-prometheus/alertmanager-operated" not found                                                                                                                        
2019/05/03 18:14:51 [ERROR] Failed to send alert: Failed to get service for alertmanager, service "cattle-prometheus/alertmanager-operated" not found 
```

Related? Or should I go ahead an open a new, separate GitHub issue for this for v2.2.3 and v2.3?

**Comment by loganhz** [comment]
@davidnuzik FYI it will be fixed in https://github.com/rancher/rancher/pull/19983

**Comment by davidnuzik** [comment]
Alerting is fixed in latest master and v2.2.3-rc6. Will give this a look now.
Expected change is that notifications now only show current event rather than past+current such as for podevents.

**Comment by davidnuzik** [comment]
Version: Master (v2.3) (5/6/19)

I can confirm that now we only get one Event Message per notification now. I tested with Pod Events. Now I will get multiple notifications with one Event Message per each. Previously these would be grouped (see comments above). We would get an event with current info then past+ current and another with past+past+current, etc.

A good example of this is when you used to create a new pod it would group the multiple pod event messages. Now they are separate notifications.
